packages/amplify-migration-e2e/src/envVariables.ts

```diff
@@ -0,0 +1,9 @@
+type EnvVariableAction = 'SET' | 'DELETE';
+
+export function toggleEnvVariable(name: string, option: EnvVariableAction, value?: string) {
+  if (option === 'SET') {
+    process.env[name] = value;
+  } else if (option === 'DELETE') {
+    delete process.env[name];
+  }
+}
```

Comment:
TBH. this doesn't seem to be providing lot of simplification over calling `process.env` directly. It's just changing syntax of how we set and unset variable without much value added.

If we want to build something that takes care of scope of env vars I'd suggest to do something like this.

```
function runWithEnvVarialbe(name: string, value: string, callable: ()=> void) => {
 process.env[name] = value;
 try {
   callable();
 } finally {
   delete process.env[name]
 }

}
```

Reply:
Sure, but I think even this wouldn't quite work, because there are a few steps which involve setting an env var, and then based on those steps running correctly, the stdout of those commands, gives another envvar which need to be set. So, if the case where I had an env var used for specific steps and then other steps use other vars, this would work well. But for this use case, I think simply going with `setEnvVariable` and `deleteEnvVariable` functions would be simple to read and understand.